.github/workflows/wheels.yml

```diff
@@ -123,4 +119,64 @@ jobs:
 
       - uses: actions/upload-artifact@v2
         with:
+          name: ${{ matrix.python }}-${{ startsWith(matrix.buildplat[1], 'macosx') && 'macosx' || matrix.buildplat[1] }}
+          path: ./wheelhouse/*.whl
+  build_universal2_wheels:
+    needs: build_wheels
+    name: Build wheel for ${{ matrix.python[0] }}-macosx-universal2
+    runs-on: macos-10.15
+    strategy:
+      # Ensure that a wheel builder finishes even if another fails
+      fail-fast: false
+      matrix:
+        python:
+          - ['cp38', '3.8']
+          - ['cp39', '3.9']
+          - ['cp310', '3.10']
+    steps:
+      - name: Checkout numpy
+        uses: actions/checkout@v2
+        with:
+          submodules: true
+          # versioneer.py requires the latest tag to be reachable. Here we
+          # fetch the complete history to get access to the tags.
+          # A shallow clone can work when the following issue is resolved:
+          # https://github.com/actions/checkout/issues/338
+          fetch-depth: 0
+      - uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python[1] }}
+          architecture: x64
+      - uses: actions/download-artifact@master
+        with:
+          name: ${{ matrix.python[0] }}-macosx
+          path: ./osx-wheels
+      - name: Fuse wheels
+        run: |
+          pip install delocate
+          wheelnames=$(find "$(pwd)/osx-wheels" -name "*.whl")
+          echo ${wheelnames[*]}
+          mkdir wheelhouse
+          delocate-fuse ${wheelnames[@]} -w wheelhouse
+          # delocate produces wheel with wrong filename
+          # rename to universal2 (FWIW, multibuild does this too)
+          wheelpath=$(find "$(pwd)/wheelhouse" -name "*.whl")
+          # Hardcoding this for now
+          # TODO: don't hardcode paths, this is tricky since we also
+          # are changing macosx SDK versions from 11 -> 10.9
+          newpath="${wheelpath/_11_0_arm64.whl/_10_9_universal2.whl}"
```

Comment:
This is hacky and very bad. I should not be hard-coding SDK versions here.
I need to get SDK versions on both x86_64 and arm64 wheels, and take the minimum version between them and then use that.

I am a big n00b at bash/sed stuff so this is going to be tricky.

Reply:
I guess the "solution" will be when we drop `<11_0` support in the far future. In the mean time, if the wheel names change this command will fail, which should fail the CI run.

Reply:
We should be able to drop `universal2` completely well before it's possible to drop macOS 10.x support.

Reply:
Good point.

Although, I guess I could set ``MACOSX_DEPLOYMENT_TARGET`` on cibuildwheel("https://cibuildwheel.readthedocs.io/en/stable/cpp_standards/#macos-and-deployment-target-versions") and re-use that value to stay on the safe side.

Reply:
I've given up on this. Its probably more effort right now than it is worth right now given that these values are unlikely to change for a while.